app/src/main/java/app/MainActivity.kt

```diff
@@ -1,12 +1,114 @@
 package app
 
 import android.os.Bundle
+import android.os.StrictMode
+import android.widget.Button
 import androidx.appcompat.app.AppCompatActivity
-import app.AnimalVilla.R
+import app.AnimalVilla.R.*
+import app.DAO.IPromptDAO
+import app.DTO.Prompt
+import org.json.JSONArray
+import org.json.JSONObject
+import java.io.BufferedReader
+import java.io.InputStreamReader
+import java.net.URL
+import java.util.concurrent.LinkedBlockingQueue
+import kotlin.concurrent.thread
+
 
 class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        setContentView(layout.activity_main)
+        findViewById<Button>(id.StartButton).setOnClickListener{
+            println("button clicked")
+            //Thread {
+              //  runOnUiThread {
+                    println(getPrompts()!!.size.toString()+" SIZE")
+
+            }
+
     }
-}
\ No newline at end of file
+    /**
+    Avoids Retrofit by self parsing
+    Returns a MutableList of prompts ordered by prompt number
+    Returns null if an error occurred
+     */
```

Comment:
Why avoid Retrofit?  This is quite a bit more code than one would need with Retrofit.
Less code is ideal.  More code has to be maintained, has more chance for errors, less flexible, etc.